### ISSUE TITLE
Sliding AnimationSliderWidget doesn't always select the right key frame

### ISSUE THREAD

**Issue by Fifourche** [napari/napari-animation#53]
If I add some key frames with different steps value to the animation, and then want to slide through it via `AnimationSliderWidget`, the key frames selected by the sliding are not consistent.

## To reproduce this :

```Python
import napari
from napari_animation import AnimationWidget
from skimage.data import cells3d

cells = cells3d()[:, 1]  # get some data

# Launch viewer to view data
viewer = napari.view_image(cells, colormap='magma')

# Instantiate the animation widget
animation_widget = AnimationWidget(viewer)
viewer.window.add_dock_widget(animation_widget, area='right')
animation = animation_widget.animation

viewer.dims.ndisplay = 3
viewer.camera.angles = (0.0, 0.0, 90.0)
animation.capture_keyframe()
viewer.camera.zoom = 2.4
animation.capture_keyframe()
viewer.camera.angles = (-7.0, 15.7, 62.4)
animation.capture_keyframe(steps=60)
viewer.camera.angles = (2.0, -24.4, -36.7)
animation.capture_keyframe(steps=60)
viewer.reset_view()
viewer.camera.angles = (0.0, 0.0, 90.0)
animation.capture_keyframe()
```

After that, sliding `AnimationSliderWidget` will reveal the problem.

**Comment by alisterburt** [comment]
Thanks for the explanation and example @Fifourche ! Very clearly broken and thanks again for the fix in #49 - let's try to get this merged 🚀

**Comment by guiwitz** [comment]
Thanks for having fixed this @Fifourche! I hadn't played with variable number of steps and so never realised the issue.

**Comment by Fifourche** [comment]
Closed by #49 !
